tests/app/services/model_records/test_model_records_sql.py

```diff
@@ -0,0 +1,210 @@
+        original_hash="111222333444",
+    )
+    store.add_model("key1", raw)
+    config1 = store.get_model("key1")
+    assert config1 is not None
+    assert config1.base == BaseModelType("sd-1")
+    assert config1.name == "model1"
+    assert config1.original_hash == "111222333444"
+    assert config1.current_hash is None
+
+
+def test_dup(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", example_config())
+    with pytest.raises(DuplicateModelException):
+        store.add_model("key1", config)
+    with pytest.raises(DuplicateModelException):
+        store.add_model("key2", config)
+
+
+def test_update(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", config)
+    config = store.get_model("key1")
+    assert config.name == "old name"
+
+    config.name = "new name"
+    store.update_model("key1", config)
+    new_config = store.get_model("key1")
+    assert new_config.name == "new name"
+
+
+def test_unknown_key(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", config)
+    with pytest.raises(UnknownModelException):
+        store.update_model("unknown_key", config)
+
+
+def test_delete(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", config)
+    config = store.get_model("key1")
+    store.del_model("key1")
+    with pytest.raises(UnknownModelException):
+        config = store.get_model("key1")
+
+
+def test_exists(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", config)
+    assert store.exists("key1")
+    assert not store.exists("key2")
+
+
+def test_filter(store: ModelRecordServiceBase):
+    config1 = DiffusersConfig(
+        path="/tmp/config1",
+        name="config1",
+        base=BaseModelType("sd-1"),
+        type=ModelType("main"),
+        original_hash="CONFIG1HASH",
+    )
+    config2 = DiffusersConfig(
+        path="/tmp/config2",
+        name="config2",
+        base=BaseModelType("sd-1"),
+        type=ModelType("main"),
+        original_hash="CONFIG2HASH",
+    )
+    config3 = VaeDiffusersConfig(
+        path="/tmp/config3",
+        name="config3",
+        base=BaseModelType("sd-2"),
+        type=ModelType("vae"),
+        original_hash="CONFIG3HASH",
+    )
+    for c in config1, config2, config3:
+        store.add_model(sha256(c.name.encode("utf-8")).hexdigest(), c)
+    matches = store.search_by_name(model_type=ModelType("main"))
+    assert len(matches) == 2
+    assert matches[0].name in {"config1", "config2"}
+
+    matches = store.search_by_name(model_type=ModelType("vae"))
+    assert len(matches) == 1
+    assert matches[0].name == "config3"
+    assert matches[0].key == sha256("config3".encode("utf-8")).hexdigest()
+    assert isinstance(matches[0].type, ModelType)  # This tests that we get proper enums back
+
+    matches = store.search_by_name(model_type=BaseModelType("sd-2"))
```

Comment:
I think this can be removed? also don't think it is quite right

Reply:
This section is rewritten. What was not quite right?

Reply:
- `matches` is set twice without its value being used
- the first time it is set, its set with `model_type=BaseModelType("sd-2")`, but `model_type` and `base_model` are different things

but I think the line can just be removed entirely

Reply:
Removed the line